sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -486,14 +486,16 @@ class Analyzer(
       case Pivot(groupByExprs, pivotColumn, pivotValues, aggregates, child) =>
         val singleAgg = aggregates.size == 1
         def outputName(value: Literal, aggregate: Expression): String = {
+          val scalaValue = CatalystTypeConverters.convertToScala(value.value, value.dataType)
+          val stringValue = Option(scalaValue).getOrElse("null").toString
```

Comment:
Maybe, I thought https://github.com/HyukjinKwon/spark/blob/3c619dfb94723bd7a7d6a0811ab6329bf107f81b/sql/core/src/test/scala/org/apache/spark/sql/DataFramePivotSuite.scala#L220-L232 covers this.

`Literal.toString` handles `null` case before. If we remove `Option(...).getOrElse("null")` there, it throws NPE in those tests.